app/assets/javascripts/govuk-component/option-select.js

```diff
@@ -0,0 +1,136 @@
+    /* This JavaScript provides two functional enhancements to option-select components:
+      1) A link that clears all of the checkboxes (referred to here as the "clearing link")
+      2) Open/closing of the list of checkboxes - this is not provided for ie6 and 7 as the performance is too janky.
+    */
+
+    this.$optionSelect = options.$el;
+    this.$options = this.$optionSelect.find("input[type='checkbox']");
+
+    // Build clearing link
+    this.$clearingLink = this.attachClearingLink();
+    this.updateClearingLink();
+
+    // Attach event listeners for clearing
+    this.$clearingLink.on('click', this.resetOptions.bind(this));
+    this.$options.on('click', this.updateClearingLink.bind(this));
+
+    // Performance in ie 6/7 is not good enough to support animating the opening/closing
+    // so do not allow option-selects to be collapsible in this case
+    var allowCollapsible = (typeof ieVersion == "undefined" || ieVersion > 7) ? true : false;
+    if(allowCollapsible){
+
+      // Add js-collapsible class to parent for CSS
+      this.$optionSelect.addClass('js-collapsible');
+
+      // Add the little arrow toggle image
+      this.attachOpenCloseToggle();
+
+      // Add open/close listeners
+      this.$optionSelect.find('.js-container-head').on('click', this.toggleOptionSelect.bind(this));
+
+      // Add key listeners the option-select is fully usable with a keyboard
+      this.$optionSelect.on('focus', this.listenForKeys.bind(this));
+      this.$optionSelect.on('blur', this.stopListeningForKeys.bind(this));
+
+      // Add a listener to the checkboxes so if you navigate to them with the keyboard you can definitely see them
+      this.$options.on('focus', this.open.bind(this));
+
+      this.close();
+    }
+  }
+
+  OptionSelect.prototype.attachClearingLink = function attachClearingLink(){
+    this.$optionSelect.find('.js-container-head').append('<a class="js-clear-selected">clear</a>');
+    return this.$optionSelect.find('.js-clear-selected');
+  };
+
+  OptionSelect.prototype.resetOptions = function resetOptions(){
+    this.$options.prop({
+      indeterminate: false,
+      "checked": false
+    }).trigger("change");
+    this.$clearingLink.addClass('js-hidden');
+
+    // Prevent the event from bubbling as there is a click handler on the parent
+    // to open/close the option-select.
+    return false;
+  };
+
+  OptionSelect.prototype.updateClearingLink = function updateClearingLink(){
+    var anyOptions = this.$options.is(":checked"),
+        clearingLinkHidden = this.$clearingLink.hasClass('js-hidden');
+
+    if (anyOptions && clearingLinkHidden) {
+      this.$clearingLink.removeClass('js-hidden');
+    } else if (!anyOptions && !clearingLinkHidden) {
+      this.$clearingLink.addClass('js-hidden');
+    }
+  };
+
+  OptionSelect.prototype.attachOpenCloseToggle = function attachOpenCloseToggle(){
+    this.$optionSelect.find('.js-container-head').append('<div class="js-toggle"></div>');
```

Comment:
Seeing as this element is only for displaying the state of the collapse do you think this element might be better named as something like `toggle-indicator`? Being called just `toggle` I was just trying to work out where the event listener to go with it was.